src/Phinx/Migration/Manager.php

```diff
@@ -368,61 +391,89 @@ public function executeSeed($name, SeedInterface $seed)
      * Rollback an environment to the specified version.
      *
      * @param string $environment Environment
-     * @param int $version
+     * @param int $target
      * @param bool $force
+     * @param bool $targetMustMatchVersionName
      * @return void
      */
-    public function rollback($environment, $version = null, $force = false)
+    public function rollback($environment, $target = null, $force = false, $targetMustMatchVersionName = true)
     {
-        $migrations = $this->getMigrations();
-        $versionLog = $this->getEnvironment($environment)->getVersionLog();
-        $versions = array_keys($versionLog);
+        // note that the migrations are indexed by name (aka creation time) in ascending order
+        $projectMigrations = $this->getMigrations();
 
-        ksort($migrations);
-        sort($versions);
+        // note that the version log are also indexed by name with the proper ascending order according to the version order
+        $executedMigrationVersions = $this->getEnvironment($environment)->getVersionLog();
```

Comment:
Inconsistent naming `$projectMigrations` vs `$executedMigrationVersions` ... very confusing when `$migrations` is right.